app/controllers/organizations_controller.rb

```diff
@@ -23,8 +23,33 @@ def create
       ).to_s
       redirect_to url_with_subdomain
     else
+      flash[:alert] = @organization.errors.full_messages.join(', ')
       render :new
-      flash[:alert] = t('organization.create.failure_message')
+    end
+  end
+
+  def login
+    # This action renders the form where the user inputs their organization site
+  end
+
+  def process_login
+    @organization = Organization.find_by(subdomain: params[:subdomain])
+    owner_role_id = Role.find_by(name: 'owner').id
+    if @organization.present?
+      subdomain = @organization.subdomain
+      host_with_subdomain = "#{subdomain}.#{APP_HOST}"
+      port = 3000
+      # Manually construct the URL with the subdomain using URI module
+      url_with_subdomain = URI::HTTP.build(
+        host: host_with_subdomain,
+        port:,
+        path: new_user_session_path, # Redirect to the login page
+        query: { role_id: owner_role_id }.to_query
+      ).to_s
+      redirect_to url_with_subdomain
```

Comment:
This is repeated code, It is also being used in `create` action. make a common method.